autopilot/scoreattach.go

```diff
@@ -0,0 +1,160 @@
+package autopilot
+
+import (
+	"bytes"
+	"sort"
+	"sync"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcutil"
+)
+
+// ScoreAttachment is an implementation of the AttachmentHeuristic
```

Comment:
Referring back to my earlier comment, I'd intended this to instead be a modification of the existing interface. So they'd _all_ be scoring based.

Reply:
Although for what we have in mind, could pin this as an `ExternalScoringHeuristic`.